modules/cudaoptflow/src/pyrlk.cpp

```diff
@@ -228,9 +235,6 @@ namespace
         CV_Assert( prevImg.channels() == 1 || prevImg.channels() == 3 || prevImg.channels() == 4 );
         CV_Assert( prevImg.size() == nextImg.size() && prevImg.type() == nextImg.type() );
 
-        // build the image pyramids.
-        buildImagePyramid(prevImg, prevPyr_, nextImg, nextPyr_, stream);
```

Comment:
Why?

Reply:
We pass, `prevPyr_` and `nextPyr_` to the `sparse` function which calculates the two pyramids if they are empty, so probably, why to pre-compute them if they are going to be computed anyways in the function. Please let me know if they are necessary here(reasons will be helpful for me to understand the insights).

See, [this](https://github.com/opencv/opencv/pull/15889/files#diff-4367b9776418cd5f0e515b2695f5805cR175) and [this](https://github.com/opencv/opencv/pull/15889/files#diff-4367b9776418cd5f0e515b2695f5805cR180) line.

Reply:
Did you run tests?
Where is assignment of `prevPyr_[0]` and `nextPyr_[0]`?

IMHO, this API should go public too:
```
PyrLKOpticalFlowBase::sparse(std::vector<GpuMat>& prevPyr, std::vector<GpuMat>& nextPyr ...)
```

Reply:
> Where is assignment of `prevPyr_[0]` and `nextPyr_[0]`?

[This line](https://github.com/opencv/opencv/pull/15889/files#diff-4367b9776418cd5f0e515b2695f5805cR64) is meant for the above purpose.



> IMHO, this API should go public too:

Working on it on my system. Will push the changes.